src/julia_internal.h

```diff
@@ -1220,11 +1221,19 @@ STATIC_INLINE void unbias_cong(uint64_t max, uint64_t *unbias) JL_NOTSAFEPOINT
 {
     *unbias = UINT64_MAX - ((UINT64_MAX % max) + 1);
 }
+
 STATIC_INLINE uint64_t cong(uint64_t max, uint64_t unbias, uint64_t *seed) JL_NOTSAFEPOINT
 {
-    while ((*seed = 69069 * (*seed) + 362437) > unbias)
-        ;
-    return *seed % max;
+    uint64_t mask = ~(uint64_t)0;
+    --max;
+    mask >>= __builtin_clzll(max|1);
```

Comment:
ideally we would be passing in the mask also  but I expect this to still be better

Reply:
true, we could attempt to keep this as the `unbias_cong` argument that it was before, but is it worth it? This seems cheap enough.